packages/rs-drive-abci/src/execution/validation/state_transition/check_tx_verification/v0/mod.rs

```diff
@@ -0,0 +1,188 @@
+                        state_transition.transform_into_action(platform, true, None)?;
+                    if !state_transition_action_result.is_valid_with_data() {
+                        return Ok(
+                            ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                                state_transition_action_result.errors,
+                            ),
+                        );
+                    }
+                    Some(state_transition_action_result.into_data()?)
+                } else {
+                    None
+                };
+
+                // Validating structure
+                let result = state_transition.validate_structure(
+                    &platform.into(),
+                    action.as_ref(),
+                    platform.state.current_protocol_version_in_consensus(),
+                )?;
+                if !result.is_valid() {
+                    return Ok(
+                        ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                            result.errors,
+                        ),
+                    );
+                }
+
+                let action = if state_transition
+                    .requires_state_to_validate_identity_and_signatures()
+                {
+                    if let Some(action) = action {
+                        Some(action)
+                    } else {
+                        let state_transition_action_result =
+                            state_transition.transform_into_action(platform, true, None)?;
+                        if !state_transition_action_result.is_valid_with_data() {
+                            return Ok(
+                                    ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                                        state_transition_action_result.errors,
+                                    ),
+                                );
+                        }
+                        Some(state_transition_action_result.into_data()?)
+                    }
+                } else {
+                    None
+                };
+
+                //
+                let result = state_transition.validate_identity_and_signatures(
+                    platform.drive,
+                    action.as_ref(),
+                    None,
+                    &mut state_transition_execution_context,
+                    platform_version,
+                )?;
+                // Validating signatures
+                if !result.is_valid() {
+                    return Ok(
+                        ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                            result.errors,
+                        ),
+                    );
+                }
+                let maybe_identity = result.into_data()?;
+
+                let action = if let Some(action) = action {
```

Comment:
Why do we need this?

Reply:
transforming into an action? because actions are transformed into events.

Reply:
Understood. We need this for fee calculation